man/tpm2-abrmd.8.in

```diff
@@ -17,11 +17,25 @@ described by the Trusted Computing Group (TGC) in the
 This daemon uses the DBus system bus and some pipes to communicate with
 clients.
 .SH OPTIONS
+.TP
+\fB\-t,\ \-\-tcti\fR
+Select the TCTI used by tabd for communication with the next component down
+the TSS stack. In most configurations this will be the TPM accessed through
+the 'device TCTI' but it could be a simulator or proxy. Only TCTIs that
+implement the common TCTI initialization mechanism are supported. Pass the
+Name of the TCTI library to load using this option. Values for this
+parameter will depend on the TCTI. Libraries are found using the same
+algorithm as dlopen (3).
+.TP
+\fB\-c,\ \-\-tcti-conf\fR
+Specify the configureation string passed to the TCTI during the 'dynamic'
+loading and initialization. The format for this string is TCTI specific.
+.TP
```

Comment:
We're going to have 2 different ways of doing this with the tools vs abrmd. Would be nice to be consistent here. Anyway we could do a tcti-loader lib in TSS to standardize this too?

tools have
```
--tcti=<name>:<conf>
```
Which helps to avoid yet-another-option in favor of a specification string. It does require names not to have a colon in them.

Reply:
Was thinking the same thing after looking at the tools just now. Was going to ask if you'd be willing to break your config string up.
```
tpm2-abrmd --tcti="libtcti-socket.so" --tcti-conf="http://127.0.0.1:2321/"
```
IMHO improves readability over:
```
tpm2-abrmd --tcti="libtcti-socket.so:http://127.0.0.1:2321/"
```
With URLs that themselves have colons etc the later just looks cluttered to me.

This is just a preference though and it doesn't trump a legit need if the tools are that pressed for letters. Should only cost me an extra call to `strtok_r` in the end.

Reply:
The major issue is pressed for letters, so I am airing on the side of less letters where possible.
However, you don't need the libtcti part, I add that on if not found from raw input, so it's as simple as:
```
tpm2-abrmd --tcti="socket:http://127.0.0.1:2321/"
```
Device example:
```
tpm2-abrmd --tcti="device:/dev/tpm0"
```
I am not married to the separator FYI. Just spit-balling here:
```
tpm2-abrmd --tcti="device(/dev/tpm0)"
```
Looks like a little function now, FWIW.

Reply:
We can probably come up with some sort of URI scheme for this: Suspend disbelief for a second and assume we're going to require people provide the absolute path to the TCTI shared object. Not  practical so we'll never really do it but it would allow us to use the `file://` prefix. With query parameters we could pass any `key, value` pair:
```
--tcti=file:///usr/local/lib/libtcti-abrmd.so?key0=value0&key1=value1&...
```
I'm not sure if the query string is allowed with the `file` schema but it's just a though experiment. We could always just start using a new sheme. Then we could just make up our own rules w/r short hand for the module file:
```
--tcti=tcti://device/?file=/dev/tpm0
--tcti=tcti://abrmd/?bus_type=session&bus_name=foo.bar
``` 
We *should* register the new schema with IANA according to wikipedia though. The only other issue I can think up is that your shortening rules may have problems on Windows. They don't prefix library file names with `lib` so a TCTI on Windows will be `tss2-foo.lib` or `tss2-foo.dll` (I think).

Reply:
Just created an issue #321 to track this. I've got another few patches queued up behind this to clean up a few things and finish off the TCTI loading / init stuff. This TCTI config encoding discussion can happen independent of this PR.

Reply:
For windows, I would need to re-write the loader, as dlopen doesn't exist. So I would change it to behave the same given the platform.

Reply:
I'm just starting to read up on this in between phone calls: https://msdn.microsoft.com/en-us/library/windows/desktop/ms684175(v=vs.85).aspx